pkg/engine/kube_test.go

```diff
@@ -0,0 +1,236 @@
+package engine
+
+import (
+	"reflect"
+	"strings"
+	"testing"
+
+	"github.com/containers/podman/v4/pkg/domain/entities"
+	"gopkg.in/yaml.v3"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func TestValidatePod(t *testing.T) {
+	var pod v1.Pod = v1.Pod{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "test",
+		},
+		TypeMeta: metav1.TypeMeta{
+			Kind:       "Pod",
+			APIVersion: "v1",
+		},
+		Spec: v1.PodSpec{
+			Containers: []v1.Container{
+				{
+					Name: "test1",
+				},
+				{
+					Name: "test2",
+				},
+			},
+		},
+	}
+
+	err := validatePod(pod)
+	if err != nil {
+		t.Errorf("expected no error, got %s", err)
+	}
+
+	pod.Spec.Containers[0].Name = pod.ObjectMeta.Name
+
+	err = validatePod(pod)
+	if err == nil {
+		t.Errorf("expected error, got nil")
+	}
+
+}
+
+var invalidYaml = `
+apiVersion: v1
+		kind: Pod
+	metadata`
+
+var notPodYaml = `
+apiVersion: v1
+kind: ConfigMap`
+
+var podYaml = `
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test_pod1
+spec:
+  containers:
+  - name: test1
+    image: test1
+---
+apiVersion: v1
+kind: Pod
+metadata:
+  name: test_pod2
+spec:
+  containers:
+  - name: test2
+    image: test2`
+
+var expectedPodList = []v1.Pod{
+	{
+		TypeMeta: metav1.TypeMeta{
```

Comment:
It is more common in k8s to omit the two embedded types - TypeMeta and ObjectMeta, eg:
```
Kind: "Pod",
APIVersion "v1",
Name: "test_pod1",
...
```